denoiser_project/Model/training.py

```diff
@@ -65,64 +74,58 @@ def denoising_autoencoder_model(input_shape):
     decoded = Conv1D(1, 3, activation=config['model']['output_activation'], padding='same')(x)
 
     autoencoder = Model(input_signal, decoded)
-
     return autoencoder
 
 
-if __name__ == '__main__':
-    # Path to the directory containing the noisy and clean audio files
-    noisy_dir = config['data']['noisy_data_path']
-    clean_dir = config['data']['clean_data_path']
-
-    os.makedirs('../Chunks', exist_ok=True)
-
-    # Directory to save the processed audio files
-    x_processed_dir = config['data']['noisy_chunk_path']
-    y_processed_dir = config['data']['clean_chunk_path']
-
-    os.makedirs(x_processed_dir, exist_ok=True)
-    os.makedirs(y_processed_dir, exist_ok=True)
-
-    all_noisy_files = os.listdir(noisy_dir)
-    all_clean_files = os.listdir(clean_dir)
-
-    # Limit to the first 1000 files
-    x_files_to_process = all_noisy_files[:1000]
-    y_files_to_process = all_clean_files[:1000]
-
-    process_files(x_files_to_process, noisy_dir, x_processed_dir)
-    process_files(y_files_to_process, clean_dir, y_processed_dir)
+def main():
+    with open('../../config.yaml', 'r') as file:
+        config = yaml.safe_load(file)
 
-    print("Processing complete!")
-
-    # Load audio files for further processing
-    noisy_audio_files = load_audio_files(x_processed_dir, sr=config['data']['sample_rate'], target_duration=config['data']['data_duration'])
-    clean_audio_files = load_audio_files(y_processed_dir, sr=config['data']['sample_rate'], target_duration=config['data']['data_duration'])
+    # Create necessary directories
+    create_directories('../Chunks', config['data']['noisy_chunk_path'], config['data']['clean_chunk_path'])
 
-    # Convert lists to numpy arrays
-    clean_data = np.array(clean_audio_files)
-    noisy_data = np.array(noisy_audio_files)
+    # Process audio files
+    noisy_data = load_and_process_data(
+        os.listdir(config['data']['noisy_data_path'])[:1000],
```

Comment:
Consider breaking down large functions into smaller, more manageable functions to enhance readability and maintainability.
all this code move to the main  function and call it as 
if __name__ == '__main__': 
        main()

Reply:
Already in main function